### ISSUE TITLE
Connect your home node to BTCPay Server via LNC

### ISSUE THREAD

**Issue by pavlenex** [btcpayserver/btcpayserver#5626]
This can be a part of the larger effort of allowing more Lightning wallets to be used as nodes in BTCPay (currently we have Alby, Blink). It seems LNC in particular works with Zeus, but beyond that it's unclear if any other wallets use it, so we should take that into consideration and potentially give more time to gathering feedback if any others potentially use this and if it's worth engineering efforts.

### Discussed in https://github.com/btcpayserver/btcpayserver/discussions/4859

<div type='discussions-op-text'>

<sup>Originally posted by **Liongrass** April  7, 2023</sup>
I'm a Lightning enthusiast (disclaimer: I also work for Lightning Labs). I run multiple node for various use cases, consuming considerable resources and capital in the process.
I also help others run Lightning nodes, including through BTCPay Server. Most commonly, the individuals and business owners I talk to, despite running their own nodes at home, dismiss BTCPay Server because of its significant resource overhead, including paying for a VPS with enough storage for a Bitcoin node, enough CPU power for a Lightning node and the time to manage liquidity. People already run their own nodes at home on their own hardware, which I think they should do! But how can they make use of BTCPay Server?

### The Remedy
Implement Lightning Node Connect (LNC) as a convenient option for people to connect their existing home node to their BTCPay Server installation on a VPS, even if their home node is behind tor. You can learn about how [LNC works under the hood here](https://docs.lightning.engineering/lightning-network-tools/lightning-terminal/lightning-node-connect), There are [npm](https://docs.lightning.engineering/lightning-network-tools/lightning-terminal/lnc-npm) and [react native](https://docs.lightning.engineering/lightning-network-tools/lightning-terminal/lnc-rn) packages that help with the integration. As BTCPay Server is almost guaranteed to run on clearnet behind a domain, it may also include [aperture](https://github.com/lightninglabs/aperture) to mitigate privacy issues that come from using an external proxy.

### Opening the path to a leaner BTCPay Server experience
Does every BTCPay Server need a Bitcoin and Lightning node inside of it? This is where the vast majority of resource overhead comes from when running your own instance. We could make it easier for people to run a BTCPay Server 'Lite', meaning without the Bitcoin or Lightning node inside. It would be ready to use almost instantly upon installation and connect to their existing node at home over LNC.

#### Share your BTCPay Server
Furthermore, such a setup would make it trivial to share your BTCPay Server installation with anyone who has a Lightning node at home. I currently use the LNbits plugin for that, but given my experiences a self-custodial solution would be superior in many ways.

#### Enhanced security with LND Accounts
LND Accounts is kind of like the LNbits plugin, but enforced by Macaroons directly by your node. Once I can connect to my own node via LNC, without any additional work, I will also be able to create a unique Account for BTCPay Server, for example limiting it to invoice-only, or segregating funds between my personal spending wallet and the funds BTCPay Server has access to. I can also create a new sub account for each of my stores, allowing for clearer accounting.

#### On-chain payments through my Lightning Node
On-chain payments might continue to be a popular request for a while, and I believe eventually they could be optionally handled through the Lightning node too. LND for instance already supports importing multiple xpubs and keep track of them in its internal wallet.

As a whole, I believe such a setup would make BTCPay Server easier to maintain, reduce cost and downtime. I would no longer need to manage liquidity for multiple nodes, juggle tailscale or SSH forwarding, I could let friends or community members use my BTCPay instance either with their home node, or a dedicated custodial account on my own node.

### Feature request
Despite the many words above, my feature request is simple. I humbly ask you to consider _LNC_ as a connection method for Lightning nodes. LNC is already available to all Umbrel, Raspiblitz, Voltage or Start9 users.</div>

**Comment by beeforbacon** [comment]
To be clear, Zeus and Alby do not provide the LNC connection. They offer LNC connection as an option for connecting your own LND node to their service for remote access through the respective apps. LNC implementation into BTCPay would likely be a additional option on the Lightning Node connection page that would say something like "LNC Pairing Phrase" and that would allow the connection from one's own LND node via clearnet regardless if the node itself is tor only. This is done through the magic of the mailbox proxy built into Lightning Node Connect. The UI would probably be something like this: 
![image](https://github.com/btcpayserver/btcpayserver/assets/51864803/857e0472-db30-4561-802a-49f4955560c5)

**Comment by pavlenex** [comment]
@beeforbacon I still don't get it, could user running Zeus with node on the phone be able to hook it into BTCPay with LNC?

**Comment by beeforbacon** [comment]
@pavlenex No, Zeus can talk to an LND node over LNC much easier and more reliably than the old (and frankly outdated) LND REST method. A good way to think about it is that Lightning Node Connect is the new alternative to LND REST with a few more cool features such as very user-friendly ways to define permissions and a few other things but at the end of the day it is just a much much better way to remotely connect your node to services, similar to LND REST but better. @dstrukt can provide a bit more detail or context on that.

**Comment by pavlenex** [comment]
Yeah but then I really don't see an end-user use-case for this one that solves an actual problem. IMO the only reason to prioritize something like this as an actual problem-solver rather than another cool thing we've built would be this

My understanding is that LDK node wallets on phone would be able to be hooked into a btcpay store.

<img width="831" alt="Screenshot 2023-12-27 at 01 32 40" src="https://github.com/btcpayserver/btcpayserver/assets/36959754/76aaa28b-d180-45a6-8a97-1ca32209d0d7">

**Comment by beeforbacon** [comment]
The big problem that it solves is it allows tor nodes to easily connect to a BTCPayServer instance, and I think that is a big deal. Further, the fact that LNC is built into every LND node now either through the LiT package or on every mynode,umbrel,start9, etc. means that this is not going away and it is going to be used more and more. Many people don't know how to go get a hex of their macaroon, etc. to get their REST connection string, and dynamic IP addresses can change any time, tls certs, etc. LNC gets rid of this huge barrier to entry and friction point, and I think that adding it to BTCPayServer as an option would be a huge step forward for a big chunk of node runners who are looking to use their own node as part of BTCPayServer.

**Comment by dstrukt** [comment]
Apologies for the delay in response during the holiday. I think @beeforbacon's response pretty much sums it up. Agree it would be a huge improvement to the current LND connection UX, and should be seriously considered.

Will confirm if there's any additional functionality planned that might satisfy the use-case @pavlenex mentioned above.

**Comment by levmi** [comment]
Thanks @dstrukt and @beeforbacon for explaining some of the details of Lightning Node Connect. Overall, think they did a fantastic job and really hit the nail on the head! There are more resources here if you want to get a better understanding of how things work [under the hood](https://docs.lightning.engineering/lightning-network-tools/lightning-terminal/lightning-node-connect) or [how to use in an application with NPM](https://docs.lightning.engineering/lightning-network-tools/lightning-terminal/lnc-npm). 

You would know way more about market sizing generally. But, like mentioned above, node operators who use Voltage, Umbrel, Start9, myNode, etc. would be able to easily "bring their own node" and connect to a BTCPayServer instance with LNC. From Lightning Terminal UI, they just generate a pairing phrase and then connect to the application. When Zeus initially enabled this, they heard from users that it was significantly smoother and more performant/persistent experience than their previous methodologies. Basically, LNC is a connection protocol for any app to connect to LND/litd node. I also like the proposed BTCPay Lite flow mentioned above, but would defer to you with regard to your market and users obviously. 

Anyway, just wanted to weigh in and reach out to let you know that we're happy to answer any questions or set up time to discuss more in-depth! Hope all is well :)

**Comment by Kukks** [comment]
Ok, I did a quick investigation. If we had to code the entire client implementation, it would involve implementing parts of BOLT8 ourselves, which is a bit (read: very) tedious. So I dug a bit more into the actual client they provide and we can get something easier by following and having this issue resolved: https://github.com/lightninglabs/lightning-node-connect/issues/71

**Comment by beeforbacon** [comment]
Anything we can do to assist with this feature?

**Comment by pavlenex** [comment]
> Anything we can do to assist with this feature? 

Hey 👋 You can do a PR, and we can review it, but our core team is prioritizing other things and this is not something we have capacity to implement in the near future .

**Comment by beeforbacon** [comment]
> > Anything we can do to assist with this feature?
> 
> Hey 👋 You can do a PR, and we can review it, but our core team is prioritizing other things and this is not something we have capacity to implement in the near future .

Thanks. I guess in the meantime here is all the info that a dev (I'm not a dev sadly, just a power user) may need to get this accomplished:

Building Lightning Apps with LNC: https://www.youtube.com/watch?v=vZWbydel-bU

LNC: Under the hood: https://docs.lightning.engineering/lightning-network-tools/lightning-terminal/lightning-node-connect
LNC demo site: https://lnc-demo.com/
LNC Guide: https://docs.lightning.engineering/lapps/guides/make-your-own-lnc-powered-application
LNC repo/libraries: https://github.com/lightninglabs/lightning-node-connect
LNC web-repo: https://github.com/lightninglabs/lnc-web
NPM LNC web module: https://www.npmjs.com/package/@lightninglabs/lnc-web
NPM LNC react module: https://www.npmjs.com/package/@lightninglabs/lnc-rn

**Comment by pavlenex** [comment]
@beeforbacon Pardon my ignorance, but is this now done with NWC integration? or is this something completely different? https://blog.getalby.com/manage-btcpay-server-with-any-node-or-wallet-with-nwc/

**Comment by beeforbacon** [comment]
Hey, @pavlenex . NWC is a completely different thing.

**Comment by Kukks** [comment]
> > > Anything we can do to assist with this feature?
> > 
> > 
> > Hey 👋 You can do a PR, and we can review it, but our core team is prioritizing other things and this is not something we have capacity to implement in the near future .
> 
> Thanks. I guess in the meantime here is all the info that a dev (I'm not a dev sadly, just a power user) may need to get this accomplished:
> 
> Building Lightning Apps with LNC: [youtube.com/watch?v=vZWbydel-bU](https://www.youtube.com/watch?v=vZWbydel-bU)
> 
> LNC: Under the hood: [docs.lightning.engineering/lightning-network-tools/lightning-terminal/lightning-node-connect](https://docs.lightning.engineering/lightning-network-tools/lightning-terminal/lightning-node-connect) LNC demo site: [lnc-demo.com](https://lnc-demo.com/) LNC Guide: [docs.lightning.engineering/lapps/guides/make-your-own-lnc-powered-application](https://docs.lightning.engineering/lapps/guides/make-your-own-lnc-powered-application) LNC repo/libraries: [lightninglabs/lightning-node-connect](https://github.com/lightninglabs/lightning-node-connect) LNC web-repo: [lightninglabs/lnc-web](https://github.com/lightninglabs/lnc-web) NPM LNC web module: [npmjs.com/package/@lightninglabs/lnc-web](https://www.npmjs.com/package/@lightninglabs/lnc-web) NPM LNC react module: [npmjs.com/package/@lightninglabs/lnc-rn](https://www.npmjs.com/package/@lightninglabs/lnc-rn)

I'd say this will be blocked until anything moves here: |
https://github.com/btcpayserver/btcpayserver/issues/5626#issuecomment-1895382796

**Comment by Fovour01** [comment]
Hello dear, How are you doing? I hope all is well with you.well I got the
email and I have issues with it but if you don't mind I will love you to
teach me how to do it I will be grateful to be in part of it.

  [ Your friend]
   Favour usunobun.

**Comment by Fovour01** [comment]
Hello dear, How are you doing? I hope all is well with you.well I got the email and I have issues with it but if you don't mind I will love you to teach me how to do it I will be grateful to be in part of it. 

  [ Your friend]
   Favour usunobun.
